src/org/abymaly/code/magnets/Magnets.java

```diff
@@ -2,4 +2,158 @@
 
 public class Magnets {
 
-}
+    public Signs[][] board;
+    private final int size;
+    public int possibleMoves = 0;
+
+    public Magnets(int size) {
+        this.size = size;
+        this.board = new Signs[size][size];
+    }
+
+    public Magnets() {
+        this.size = 4;
+    }
+
+    class StartPosition {
+        int x;
+        int y;
+
+        public StartPosition() {
+        }
+    }
+
+    StartPosition startPosition = new StartPosition();
+
+    public void printBoard(Signs[][] board) {
+        String newLine = "\n";
+        StringBuffer sbLine = new StringBuffer();
+
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                if (board[i][j].equals(Signs.EMPTY)) {
+                    sbLine.append(" _ ");
+                } else {
+                    sbLine.append(" ");
+                    sbLine.append(board[i][j]);
+                    sbLine.append(" ");
+                }
+            }
+            sbLine.append(newLine);
+        }
+        System.out.println(sbLine);
+    }
+
+    public void setStartPosition(Signs[][] board, int x, int y) {
+        board[x][y] = Signs.O;
+    }
+
+    public void setData(Signs[][] board, int x, int y, Signs value) {
+        board[x][y] = value;
+    }
+
+    public Signs[][] emptyBoard () {
+        Signs [][] newBoard = new Signs[size][size];
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                setData(newBoard, i, j, Signs.EMPTY);
+            }
+        }
+        return newBoard;
+    }
+
+    public void refreshStartPosition(Signs[][] board) {
+        setStartPosition(board, startPosition.x, startPosition.y);
+    }
+
+    public void emptyStartPosition (Signs[][] board) {
+        setData(board, startPosition.x, startPosition.y, Signs.EMPTY);
+    }
+
+    public void getStartPosition(Signs[][] board) {
```

Comment:
Mam wrażenie, że to jest powtórzenie wzorca z kalendarza.
Z wyglądu planszy wynika położenie kulki, a położenie kulki by się dało łatwo ustalić tymi wspomnianym zmiennymi `x` i `y`.
Ja tu widzę taki kod:
```Java
void move( int newX, int newY) {
  setBoard(x,y,EMPTY)
  x=newX
  y=newY
  setBoard(x,y,Sign.O)
```
I wtedy nie musisz szukać na planszy, bo masz od ręki.
Było to trudno napisać, bo `board` nie jest elementem klasy - skutek pierwszej uwagi.